Kickstarter-iOS/Views/Controllers/DiscoveryViewControllerTests.swift

```diff
@@ -0,0 +1,32 @@
+@testable import Kickstarter_Framework
+import XCTest
+
+internal final class DiscoveryViewControllerTests: TestCase {
+  func testScrollToTop_DiscoveryPages() {
```

Comment:
Can I suggest this change:

```swift
func testScrollToTop_DiscoveryPages() {
  let vc = DiscoveryViewController.instantiate()
  _ = vc.view

  let configureExpectation = expectation(description: "Waiting for view controllers to be configured")
  let pageViewController = vc.children
    .compactMap { $0 as? UIPageViewController }.first

  DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
    while pageViewController?.viewControllers?.first?.view as? UIScrollView != nil {
      configureExpectation.fulfill()
      break
    }
  }

  waitForExpectations(timeout: 10, handler: nil)

  XCTAssertNotNil(pageViewController?.viewControllers?.first?.view as? UIScrollView)
}
```

The main difference is that the test will take as little as 0.1 seconds but not more than 10 seconds. I think it _might_ be less flakey but in most cases will pass quickly.

You can repurpose this for the other async test too.

Reply:
Still won't pass :/